### DISCUSSION TITLE
find PRs that update a file

### DISCUSSION THREAD

**Discussion by edwardchenchen** [community/community#36946] [Pull Requests]
when I am going through a repo and I find a bug in a file or a file in a doc repo is old
before I make contributions I wanted to see if anyone has already opened a PR for this file
but going through all the PR is not feasible, especially in large repos with 100+ open PR.

Is there currently a way to find PRs that update a file?

**Comment by grahamperrin** [comment]
> … Is there currently a way to find PRs that update a file?

As far as I can tell: no, not with the usual web interface to GitHub. 

Accepted answer <https://webapps.stackexchange.com/a/104839/224355> offered an example: 

    is:pr is:open "phrase to search for"

A nice idea, however it's not a reliable approach. 

For example: 

* <https://github.com/freebsd/freebsd-quarterly/pulls?q=is%3Apr+cloud-init> does find a pull request for a file
* specifying the exact name of the one file in the PR <https://github.com/freebsd/freebsd-quarterly/pulls?q=is%3Apr+cloud-init.adoc> fails to find the PR. 

Elsewhere in Stack Exchange: [Can I get a list of all pull requests related to a specific file/directory path in GitHub? - Stack Overflow](https://stackoverflow.com/questions/42496939/can-i-get-a-list-of-all-pull-requests-related-to-a-specific-file-directory-path) – the only answer uses [git, not GitHub](https://stackoverflow.com/a/42498685/38108). 

### Reference

[Searching issues and pull requests - GitHub Docs](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests)

**Comment by ossanna16** [comment]
Hi there @edwardchenchen and welcome to our community 😄. Thanks so much for asking a great question!

**Comment by TWiStErRob** [comment]
2 years and 1 major "[search rewrite](https://github.blog/2023-02-06-the-technology-behind-githubs-new-code-search/)" later:

> The path qualifier is not supported when searching pull requests.

![image](https://github.com/community/community/assets/2906988/99f5dec3-0877-4b6a-84e6-1a7303b7ccee)

It was a promising idea though :(

**Comment by karimmacdonald** [comment]
I just needed to do this. The GitHub CLI (`brew install gh`) makes it easy: here's a quick-n-dirty 1-liner:
```
PAGER= gh pr list | cut -f 1 -w | while read l; do echo ${l}; PAGER= gh pr diff --name-only ${l} | grep my-filename ; done
```
